pkg/eksd/upgrader.go

```diff
@@ -0,0 +1,40 @@
+package eksd
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/logger"
+	"github.com/aws/eks-anywhere/pkg/types"
+)
+
+func (e *Installer) Upgrade(ctx context.Context, cluster *types.Cluster, currentSpec, newSpec *cluster.Spec) (*types.ChangeDiff, error) {
```

Comment:
Was this supposed to be `(u *Upgrader)`?

Reply:
I will separate the two interfaces in my follow-up PR.